### PR TITLE
[ONNX] Move graph transform functions to 'passes'

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #95651
* #95935
* #95929
* __->__ #95664
* #95676
* #95686
* #95650

This PR only moved code to their new location. No other actual code changes.

### CODE REVIEW

File: torch/onnx/_internal/fx/passes/fx_to_onnxscript.py [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+
+
+@_beartype.beartype
+def _validate_op_between_ort_torch(
+    node: torch.fx.Node,
+    symbolic_fn: Union[onnxscript.OnnxFunction, Callable],
+    torch_args: tuple,
+    torch_kwargs: dict,
+):
+    """Validate the op between ONNX Runtime and PyTorch."""
```
Comment (wschin) [inline torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:444]: What does this function really do? The doc string is basically the same as its name.

File: torch/onnx/_internal/fx/passes/fx_to_onnxscript.py [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+        with evaluator.default_as(evaluator.ort_evaluator):
+            expected_outputs = node.target(*torch_args, **torch_kwargs)  # type: ignore[operator]
+            # TODO(titaiwang): Expose _convert_tensor_to_numpy and _convert_kwargs_for_onnx?
+            input_onnx = [
+                onnx_proto_utils._convert_tensor_to_numpy(x) for x in torch_args
+            ]
+            kwargs_onnx = _filter_incompatible_and_dtype_convert_kwargs(torch_kwargs)
+            ort_outputs = symbolic_fn(*input_onnx, **kwargs_onnx)
+
+            # TODO: add pytree structure comparison.
```
Comment (wschin) [inline torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:463]: Impossible to compare pytree structure. FX allows recursive collections such as `List[Dict[List[.....]`.

File: torch/onnx/_internal/fx/passes/fx_to_onnxscript.py (hunk 1 above)
Comment (BowenBao) [reply torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:444]: This PR only splits code from one file to several.

File: torch/onnx/_internal/fx/passes/fx_to_onnxscript.py (hunk 2 above)
Comment (BowenBao) [reply torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:463]: Could you elaborate what does `recursive collections` mean? Is it just nested containers, or are you suggesting it's possible to have cycles?

And could you elaborate why such structure makes it impossible to compare?

File: torch/onnx/_internal/fx/passes/fx_to_onnxscript.py (hunk 1 above)
Comment (titaiwangms) [reply torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:444]: This function is trying to validate op-by-op results with runtime shape generated inputs to provide op-level debugging. I have a plan to refactor this once we have settled the shape inference of graph (Currently, we don't have shape in proto and I haven't figured out how dynamic axes would affect the shape usage in this function, but this is default off).

File: torch/onnx/_internal/fx/passes/fx_to_onnxscript.py (hunk 2 above)
Comment (titaiwangms) [reply torch/onnx/_internal/fx/passes/fx_to_onnxscript.py:463]: Reopen the function related issue that we could have more talk about this: https://github.com/microsoft/onnx-script/issues/393

**Review by justinchuby — APPROVED** [review]

**Pull request closed without merging**